template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kafka/KafkaMultiAzConfigProvider.java

```diff
@@ -54,7 +54,8 @@ public String getServiceType() {
 
     @Override
     public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
-        return (StackType.DATALAKE.equals(source.getStackType()) || StackType.WORKLOAD.equals(source.getStackType()))
+        return (StackType.DATALAKE.equals(source.getStackType()) || StackType.WORKLOAD.equals(source.getStackType()) ||
+            StackType.TEMPLATE.equals(source.getStackType()))
```

Comment:
Sorry but this does not make sense. The stack of clusters cannot be of `StackType.TEMPLATE`. The latter represents a customer cluster definition, that is called a "cluster template" in the CB repo for historical reasons (Ambari times). The stack is either `DATALAKE` or `WORKLOAD` (for DH clusters), even if they are launched from a custom cluster definition (that is internally stored as a stack of `StackType.TEMPLATE`). Adding this extra condition will not solve your issues.